.github/workflows/build-wheel-linux-arm64.yaml

```
@@ -37,291 +34,415 @@ jobs:
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/mlir/Enzyme
+          key: enzyme-${{ needs.constants.outputs.enzyme_version }}-container-source
+          enableCrossOsArchive: True
+          fail-on-cache-miss: True
+
+      - name: Get Cached Enzyme Build
+        id: cache-enzyme-build
+        uses: actions/cache/restore@v4
+        with:
+          path: ${{ github.workspace }}/enzyme-build
+          key: ${{ matrix.container_img }}-enzyme-${{ needs.constants.outputs.llvm_version }}-${{ needs.constants.outputs.enzyme_version }}-wheel-build
+          fail-on-cache-miss: True
+
+      - name: Build Catalyst-Runtime
+        run: |
+          cmake -S runtime -B $GITHUB_WORKSPACE/runtime-build -G Ninja \
+                -DCMAKE_BUILD_TYPE=Release \
+                -DCMAKE_LIBRARY_OUTPUT_DIRECTORY="$GITHUB_WORKSPACE/runtime-build/lib" \
+                -DPython_EXECUTABLE=$(which python${{ matrix.python_version }}) \
+                -DENABLE_OPENQASM=ON \
+                -DENABLE_OQD=OFF \
+                -DMLIR_INCLUDE_DIRS="$GITHUB_WORKSPACE/mlir/llvm-project/mlir/include" \
+                -DCMAKE_CXX_FLAGS="-Wno-error=maybe-uninitialized"
+
+          cmake --build $GITHUB_WORKSPACE/runtime-build --target rt_capi rtd_openqasm rtd_null_qubit
+
+      - name: Build OQC-Runtime
+        run: |
+            C_COMPILER=$(which gcc) \
+            CXX_COMPILER=$(which g++) \
+            OQC_BUILD_DIR="$GITHUB_WORKSPACE/oqc-build" \
+            RT_BUILD_DIR="$GITHUB_WORKSPACE/runtime-build" \
+            PYTHON=$(which python${{ matrix.python_version }}) \
+            make oqc
+
+      # Build Quantum and Gradient Dialects
+      - name: Build MLIR Dialects
+        run: |
+          export PATH=$GITHUB_WORKSPACE/llvm-build/bin:$PATH
+          cmake -S mlir -B $GITHUB_WORKSPACE/quantum-build -G Ninja \
+                -DCMAKE_BUILD_TYPE=Release \
+                -DLLVM_ENABLE_ASSERTIONS=ON \
+                -DQUANTUM_ENABLE_BINDINGS_PYTHON=ON \
+                -DPython3_EXECUTABLE=$(which python${{ matrix.python_version }}) \
+                -DPython3_NumPy_INCLUDE_DIRS=$(python${{ matrix.python_version }} -c "import numpy as np; print(np.get_include())") \
+                -DMLIR_DIR="$GITHUB_WORKSPACE/llvm-build/lib/cmake/mlir" \
+                -DMHLO_DIR="$GITHUB_WORKSPACE/mhlo-build/lib/cmake/mlir-hlo" \
+                -DMHLO_BINARY_DIR="$GITHUB_WORKSPACE/mhlo-build/bin" \
+                -DEnzyme_DIR="$GITHUB_WORKSPACE/enzyme-build" \
+                -DENZYME_SRC_DIR="$GITHUB_WORKSPACE/mlir/Enzyme" \
+                -DLLVM_ENABLE_ZLIB=FORCE_ON \
+                -DLLVM_ENABLE_ZSTD=OFF \
+                -DLLVM_ENABLE_LLD=ON
+
+          cmake --build $GITHUB_WORKSPACE/quantum-build --target check-dialects catalyst-cli
+
+      - name: Build Plugin wheel
+        # Run only on Thursday at the given time
+        if: github.event.schedule == '35 4 * * 4'
```

Could just not check the event_name instead?

```suggestion
        if: github.event_name == "schedule"
```

---

This is just copied over from our actions, probably your suggestion is a good one.

But the schedule is missing from this workflow entirely so we should probably add it back in:
```
  schedule:
    # Thursdays we test the standalone plugin
    - cron: '35 4 * * 4'
 ```

---

i've moved it to scheduled